autoload/utils/exec/term.vim

```diff
@@ -0,0 +1,123 @@
+" autoload/utils/exec/term.vim - contains executable helpers
+" Maintainer:   Ilya Churaev <https://github.com/ilyachur>
+
+" Private functions {{{ "
+let s:cmake4vim_term = {}
+let s:err_fmt = ''
+
+function! s:createQuickFix() abort
+    let l:old_error = &errorformat
+    if s:err_fmt !=# ''
+        let &errorformat = s:err_fmt
+    endif
+    cgetexpr join(s:cmake4vim_term['cout'], "\n")
+    silent call setqflist( [], 'a', { 'title' : s:cmake4vim_term[ 'cmd' ] } )
+    if s:err_fmt !=# ''
+        let &errorformat = l:old_error
+    endif
+    " Remove cmake4vim job
+    let s:cmake4vim_term = {}
+endfunction
+
+function! s:prepareOut(msg) abort
+    let l:without_ascii = substitute(a:msg, '\%x1B\[[0-9;]*\a', '', 'g')
+    let l:lines = split(l:without_ascii, '\r')
+    return l:lines
+endfunction
+
+" Vim functions {{{ "
+function! s:vimOut(channel, msg) abort
+    let s:cmake4vim_term['cout'] += s:prepareOut(a:msg)
+endfunction
+
+function! s:vimClose(channel, status) abort
+    let l:open_qf = get(s:cmake4vim_term, 'open_qf', 0)
+
+    let l:cmd = s:cmake4vim_term['cmd']
+    call s:createQuickFix()
+
+    if l:open_qf == 0
+        silent cwindow
+    else
+        silent copen
+    endif
+    cbottom
+
+    if a:status == 0
+        echon 'Success! ' . l:cmd
+    else
+        echon 'Failure! ' . l:cmd
+    endif
+endfunction
+" }}} Vim functions "
+
+" nvim functions {{{ "
+function! s:nVimOut(job_id, data, event) abort
+    for val in filter(a:data, '!empty(v:val)')
+        let s:cmake4vim_term['cout'] += s:prepareOut(val)
+    endfor
+endfunction
+
+function! s:nVimExit(job_id, data, event) abort
+    if empty(s:cmake4vim_term) || a:job_id != s:cmake4vim_term['job']
+        return
+    endif
+    let l:job = s:cmake4vim_term['job']
+    let l:cmd = s:cmake4vim_term['cmd']
+
+    let l:open_qf = get(s:cmake4vim_term, 'open_qf', 0)
+    silent exec 'bwipeout! ' . s:cmake4vim_term['termbuf']
+
+    call s:createQuickFix()
+
+    if a:data != 0 || l:open_qf != 0
+        copen
+    endif
+    " if a:data == 0
+    "     echon 'Success! ' . l:cmd
+    " else
+    "     echon 'Failure! ' . l:cmd
+    " endif
+endfunction
+" }}} nvim functions "
+" }}} Private functions "
+
+function! utils#exec#term#run(cmd, open_qf, err_fmt) abort
+    let l:cmake4vim_term = 'cmake4vim_execute'
```

Comment:
This can be declared after the `if` or not at all since it's used only once.

Reply:
Yes, I refactor this code.